File: 0228_two_pointers.js
/**
 * @param {number[]} nums
 * @return {string[]}
 */
var summaryRanges = function(nums) {
    const array = new Array();
    let p1 = 0;
    let p2 = 0;
    for (let i = 0; i < nums.length; ++i) {
        if (nums[i] + 1 === nums[i + 1]) {
            ++p2;
        } else {
            if (p1 !== p2) {
                array.push(`${nums[p1]}->${nums[i]}`);     
            } else {
                array.push(`${nums[i]}`);
            }
            p1 = ++p2;
        } 
    }
    return array;
};

File: 0228_two_pointers.rb
# @param {Integer[]} nums
# @return {String[]}
def summary_ranges(nums)
    array = []
    p1 = 0
    p2 = 0

    nums.each_with_index do |num, i| 
        if num + 1 == nums[i + 1]
            p2 += 1
        else
            if p1 != p2
                array << "#{nums[p1]}->#{nums[i]}"
            else 
                array << "#{nums[i]}"
            end
            p1 = p2 += 1
        end
    end
    return array
end
